lib/partitioninfo.coffee

```diff
@@ -57,24 +57,30 @@ getPartitions = (disk, offset, getLogical = true) ->
 					result
 		result
 
-getPartitionFromList = (partitions, number) ->
-	partition = partitions[number - 1]
-	if not partition
-		throw new Error("Partition not found: #{number}.")
-	partition
+partitionNotFoundError = (number) ->
+	throw new Error("Partition not found: #{number}.")
 
-get = (disk, definition) ->
+get = (disk, number) ->
+	if number < 1
+		throw new Error('The partition number must be at least 1.')
 	getPartitions(disk, 0, false)
 	.then (partitions) ->
-		primary = getPartitionFromList(partitions, definition.primary)
-		if not definition.logical
-			primary
-		else if not MBR.Partition.isExtended(primary.type)
-			throw new Error("Not an extended partition: #{definition.primary}.")
+		position = number - 1
+		if partitions.length == 0
+			partitionNotFoundError(number)
+		else if position < partitions.length
+			partitions[position]
 		else
-			getLogicalPartitions(disk, primary.offset, primary.offset, definition.logical - 1)
+			[..., last] = partitions
+			if not MBR.Partition.isExtended(last.type)
```

Comment:
Hmmm, this seems to be assuming that the Extended partition (if any) is always the last Primary partition on the disk. While that may be "common practice", AFAIK the Extended partition could actually be _any_ one of the Primary partitions, with other Primary partitions occurring after it?
i.e. you might have
```
|  Primary 1  |  Primary 2 (Extended)                   |  Primary 3  |
|  {primary}  |  { Logical 1 | Logical 2 | Logical 3 }  |  {primary}  |
```

(I have no idea if you're allowed to have more than one Primary partition marked as Extended though)